build_tools/benchmarks/collect_compilation_statistics.py

```diff
@@ -0,0 +1,194 @@
+]
+
+
+def match_module_cmake_target(module_path: str) -> Optional[str]:
+  # Get the last 4 parts of module path. They are expected to be:
+  # benchmark_suites/<category>/vmfb/<module filename>.vmfb
+  path_parts = PurePath(module_path).parts[-4:]
+  if len(path_parts) < 4:
+    return None
+  if path_parts[0] != BENCHMARK_SUITE_REL_PATH:
+    return None
+  if path_parts[2] != MODULE_DIR:
+    return None
+  if os.path.splitext(path_parts[3])[1] != MODULE_FILE_EXTENSION:
+    return None
+  return os.path.join(*path_parts)
+
+
+def parse_compilation_time_from_ninja_log(log_path: str) -> Dict[str, int]:
+  """Retrieve the compilation time from the Ninja build log."""
+
+  target_build_time_map = {}
+  with open(log_path, "r") as f:
+    header = f.readline()
+    if NINJA_LOG_HEADER not in header:
+      raise NotImplementedError(f"Unsupported ninja log version: {header}")
+
+    for line in f:
+      start_time, end_time, _, target, _ = line.strip().split("\t")
+      cmake_target = match_module_cmake_target(target)
+      if cmake_target is None:
+        continue
+
+      start_time = int(start_time)
+      end_time = int(end_time)
+      target_build_time_map[cmake_target] = end_time - start_time
+
+  return target_build_time_map
+
+
+def get_module_component_info(module_path: str) -> ModuleComponentSizes:
+  module_zipinfo = zipfile.ZipFile(module_path)
+  size_map = dict(
+      (info.filename, info.file_size) for info in module_zipinfo.infolist())
+  vm_component_size = size_map[VM_COMPONENT_NAME]
+  const_component_size = size_map[CONST_COMPONENT_NAME]
+  identified_names = {VM_COMPONENT_NAME, CONST_COMPONENT_NAME}
+  total_dispatch_component_size = 0
+  for filename, size in size_map.items():
+    for pattern in DISPATCH_COMPONENT_PATTERNS:
+      if re.match(pattern, filename):
+        total_dispatch_component_size += size
+        identified_names.add(filename)
+        break
+
+  if identified_names != set(size_map.keys()):
+    raise AssertionError(
+        f"Unrecognized components in the module: {size_map.keys()}.")
+
+  return ModuleComponentSizes(
+      file_size=os.stat(module_path).st_size,
+      vm_component_size=vm_component_size,
+      const_component_size=const_component_size,
+      total_dispatch_component_size=total_dispatch_component_size)
+
+
+def get_module_path(benchmark_case_dir: str) -> str:
+  """Retrieve the module path for compilation statistics from the flag file."""
+
+  flagfile_path = os.path.join(benchmark_case_dir, BENCHMARK_FLAGFILE)
+  module_path = None
+  with open(flagfile_path, "r") as f:
+    for line in f:
+      match = re.match("--module_file=(.+)", line.strip())
+      if match:
+        module_name, module_ext = os.path.splitext(match.group(1))
+        module_path = f"{module_name}-{COMPILATION_STATS_MODULE_SUFFIX}{module_ext}"
+        break
+
+  if not module_path:
+    raise AssertionError(
+        f"Can't find the module file in the flagfile: {flagfile_path}")
```

Comment:
Same. This should not be an AssertionError. Also the AssertionError in main below

Reply:
I didn't find a good built-in exception, use the generic RuntimeException instead.